main.go

```diff
@@ -251,31 +232,22 @@ func ICMPReceiver(done <-chan struct{}, af string) (chan interface{}, error) {
 
 	go func() {
 		// TODO: remove hardcode; 20 bytes for IP header, 8 bytes for ICMP header, 8 bytes for TCP header
-		packet := make([]byte, outerIPHdrSize+icmpHdrSize+innerIPHdrSize+tcpHdrSize)
+		packet := make([]byte, icmpHdrSize+maxIP4HeaderSize+maxTCPHdrSize)
 		for {
-			n, from, err := syscall.Recvfrom(recvSocket, packet, 0)
+			n, from, err := conn.ReadFrom(packet)
 			if err != nil {
 				break
 			}
 			// extract the 8 bytes of the original TCP header
-			if n < outerIPHdrSize+icmpHdrSize+innerIPHdrSize+tcpHdrSize {
+			if n < icmpHdrSize+minInnerIPHdrSize+minTCPHdrSize {
 				continue
 			}
 			// not ttl exceeded
-			if packet[outerIPHdrSize] != icmpMsgType || packet[outerIPHdrSize+1] != 0 {
+			if packet[0] != icmpMsgType || packet[1] != 0 {
```

Comment:
This is a packet read from a `icmp.PacketConn`. [`ReadFrom`](https://godoc.org/golang.org/x/net/icmp#PacketConn.ReadFrom) gives us the ICMP message without IP headers.